presto-docs/src/main/sphinx/connector/hive.rst

```diff
@@ -191,6 +188,194 @@ This table can then be queried in Presto::
+
+
+``hive.max-initial-splits``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+ * **Type:** ``Integer``
+ * **Default value:** ``200``
+ * **Description:** This property describes how many splits may be initially created for a single query. The initial splits are created to allow better concurrency for small queries. Hive connector will create first ``hive.max-initial-splits`` splits with size of ``hive.max-initial-split-size`` instead of ``hive.max-split-size``. Having this value higher will force more splits to have smaller size effectively increasing definition of what is considered small query in database.
+
+
+``hive.max-outstanding-splits``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+ * **Type:** ``Integer`` (at least ``1``)
+ * **Default value:** ``1000``
+ * **Description:** Limit of number of splits waiting to be served by split source. After reaching this limit writers will stop writing new splits to split source until some of them are used by workers. Higher value will increase memory usage, but will allow to concentrate all IO at one time which may be much faster and increase resources utilization.
+
+
+``hive.max-partitions-per-writers``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+ * **Type:** ``Integer`` (at least ``1``)
+ * **Default value:** ``100``
+ * **Description:** Maximum number of partitions per writer. If higher number of partitions per writer will be required to complete query, the query will fail. By manipulating this value one may change how large queries are meant to be dropped from DB which may help with error detection.
+
+
+``hive.max-split-iterator-threads``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+ * **Type:** ``Integer`` (at least ``1``)
+ * **Default value:** ``1000``
+ * **Description:** This property describes how many threads may be used to iterate through splits when loading them to the worker nodes. Higher value may increase parallelism, but high concurrency may cause time being wasted on context switching.
+
+
+``hive.max-split-size``
+^^^^^^^^^^^^^^^^^^^^^^^
+
+ * **Type:** ``String`` (data size)
+ * **Default value:** ``64 MB``
+ * **Description:** This value describes max size of split that is created after using all ``hive-max-initial-split-size`` of initial splits. The logic of initial splits is described in ``hive.max-initial-splits``. Having this value higher causes smaller parallelism which may be desirable when queries are very large and cluster is stable allowing to process data locally more efficiently without wasting time for context switching, synchronization and data collecting. The optimal value should be aligned with average query size in system.
```

Comment:
The maximum size of splits created after the initial splits.  The logic for initial splits is described in ``hive.max-initial-splits``.  A higher values will reduce parallelism.  This may be desirable for very large queries and a stable cluster because it allows for more efficient processing of local data without  context switching, synchronization and data collection that result from parallelization.  The optimal value should be aligned with the average query size in the system.